### PR TITLE
Implement data structures for Camera and Transform

### PR DESCRIPTION
**Pull request by sr229**
Implements the required data structures for Camera and Entity Transforms.

### CODE REVIEW

**Comment by sr229** [comment]
Blocked until @LeNitrous finishes Veldrid Graphics API

File: source/engine/Sekai.Engine/Entity.cs [hunk 1]
```diff
@@ -29,6 +29,11 @@ public IEnumerable<Entity> Entities
         }
     }
 
+    /// <summary>
+    /// The transform of the entity
+    /// </summary>
+    public Transform? Transform { get; set; }
```
Comment (LeNitrous) [inline source/engine/Sekai.Engine/Entity.cs:35]: The entity should never have any references to any components. They should only be obtained via `GetComponent<T>()` or similar methods.

**Review by LeNitrous — CHANGES_REQUESTED** [review]

File: source/engine/Sekai.Engine/Entity.cs (hunk 1 above)
Comment (sr229) [reply source/engine/Sekai.Engine/Entity.cs:35]: How do we plan to apply transform on a entity if we don't keep a reference in the object?

Comment (LeNitrous) [reply source/engine/Sekai.Engine/Entity.cs:35]: there will be entity processors responsible for accessing components of a given type

**Review by LeNitrous — APPROVED** [review]

**Pull request merged**